src/raven.js

```diff
@@ -469,6 +469,30 @@ Raven.prototype = {
         return this;
     },
 
+    /*
+     * Adds a shouldSendCallback callback that chains after any previously set
+     * callback(s).
+     *
+     * @param {function} callback The callback to run which allows
+     *                            introspecting the blob before sending
+     * @return {Raven}
+     */
+    addShouldSendCallback: function(callback) {
+        var prevCallback = this._globalOptions.shouldSendCallback;
```

Comment:
I'm writing a small plugin to solve https://github.com/getsentry/raven-js/issues/279. This API method would be useful.